source/Cosmos.System2/Graphics/SVGAIICanvas.cs

```diff
@@ -1,398 +1,169 @@
+using System.Text;
+using Cosmos.Debug.Kernel;
+using Cosmos.HAL.Drivers.PCI.Video;
+
+namespace Cosmos.System.Graphics
+{
+    public class SVGAIICanvas : Canvas
+    {
+        // debugger
+        internal Debugger debugger = new Debugger("System", "SVGAIICanvas");
+
+        // driver
+        private VMWareSVGAII driver;
+
+        // available modes
+        public override List<Mode> AvailableModes { get; } = new List<Mode>()
+        {
+            new Mode(320, 200, ColorDepth.ColorDepth32),
+            new Mode(320, 240, ColorDepth.ColorDepth32),
+            new Mode(512, 384, ColorDepth.ColorDepth32),
+            new Mode(640, 480, ColorDepth.ColorDepth32),
+            new Mode(720, 480, ColorDepth.ColorDepth32),
+            new Mode(800, 600, ColorDepth.ColorDepth32),
+            new Mode(1024, 768, ColorDepth.ColorDepth32),
+            new Mode(1152, 768, ColorDepth.ColorDepth32),
+
+            /* Old HD-Ready Resolutions */
+            new Mode(1280, 720, ColorDepth.ColorDepth32),
+            new Mode(1280, 768, ColorDepth.ColorDepth32),
+            new Mode(1280, 800, ColorDepth.ColorDepth32),  // WXGA
+            new Mode(1280, 1024, ColorDepth.ColorDepth32), // SXGA
+
+            /* Better HD-Ready Resolutions */
+            new Mode(1360, 768, ColorDepth.ColorDepth32),
+            //new Mode(1366, 768, ColorDepth.ColorDepth32),  // Original Laptop Resolution - this one is for some reason broken in vmware
+            new Mode(1440, 900, ColorDepth.ColorDepth32),  // WXGA+
+            new Mode(1400, 1050, ColorDepth.ColorDepth32), // SXGA+
+            new Mode(1600, 1200, ColorDepth.ColorDepth32), // UXGA
+            new Mode(1680, 1050, ColorDepth.ColorDepth32), // WXGA++
+
+            /* HDTV Resolutions */
+            new Mode(1920, 1080, ColorDepth.ColorDepth32),
+            new Mode(1920, 1200, ColorDepth.ColorDepth32), // WUXGA
+
+            /* 2K Resolutions */
+            new Mode(2048, 1536, ColorDepth.ColorDepth32), // QXGA
+            new Mode(2560, 1080, ColorDepth.ColorDepth32), // UW-UXGA
+            new Mode(2560, 1600, ColorDepth.ColorDepth32), // WQXGA
+            new Mode(2560, 2048, ColorDepth.ColorDepth32), // QXGA+
+            new Mode(3200, 2048, ColorDepth.ColorDepth32), // WQXGA+
+            new Mode(3200, 2400, ColorDepth.ColorDepth32), // QUXGA
+            new Mode(3840, 2400, ColorDepth.ColorDepth32), // WQUXGA
+        };
+
+        /// <summary>
+        /// Default graphics mode - 640x480x32
+        /// </summary>
+        public override Mode DefaultGraphicMode { get; } = new Mode(640, 480, ColorDepth.ColorDepth32);
+
+        /// <summary>
+        /// Get and set active video mode
+        /// </summary>
+        private Mode mode;
+        public override Mode Mode
+        {
+            get { return mode; }
+            set
+            {
+                mode = value;
+                if (SendDebug) { debugger.SendInternal($"Set mode property to {value}"); }

```

Comment:
I dont really see the need for SendDebug. Most if not all debug statements should be able to be removed once the code has been finished. The rest can be toggled via COSMOSDEBUG. So I think its better if you remove them.